Spigot-API-Patches/0199-Add-villager-reputation-API.patch

```diff
@@ -0,0 +1,161 @@
++        Preconditions.checkNotNull(reputation, "reputation cannot be null");
++
++        this.reputation = reputation;
++    }
++
++    /**
++     * Gets the reputation value for a specific {@link ReputationType}.
++     *
++     * @param type The {@link ReputationType type} of reputation to get.
++     * @return The value of the {@link ReputationType type}.
++     */
++    public int getReputation(@NotNull ReputationType type) {
++        Preconditions.checkNotNull(type, "the reputation type cannot be null");
++        return reputation.getOrDefault(type, 0);
++    }
++
++    /**
++     * Sets the reputation value for a specific {@link ReputationType}.
++     *
++     * @param type The {@link ReputationType type} of reputation to set.
++     * @param value The value of the {@link ReputationType type}.
++     */
++    public void setReputation(@NotNull ReputationType type, int value) {
++        Preconditions.checkNotNull(type, "the reputation type cannot be null");
++        reputation.put(type, value);
++    }
++}
+diff --git a/src/main/java/com/destroystokyo/paper/entity/villager/ReputationType.java b/src/main/java/com/destroystokyo/paper/entity/villager/ReputationType.java
+new file mode 100644
+index 00000000..5600fcdc
+--- /dev/null
++++ b/src/main/java/com/destroystokyo/paper/entity/villager/ReputationType.java
+@@ -0,0 +1,36 @@
++package com.destroystokyo.paper.entity.villager;
++
++/**
++ * A type of reputation gained with a {@link org.bukkit.entity.Villager Villager}.
++ * <p>
++ * All types but {@link #MAJOR_POSITIVE} are shared to other villagers.
++ */
++public enum ReputationType {
++    /**
++     * A gossip with a majorly negative effect. This is only gained through killing a nearby
++     * villager.
++     */
++    MAJOR_NEGATIVE,
++
++    /**
++     * A gossip with a minor negative effect. This is only gained through damaging a villager.
++     */
++    MINOR_NEGATIVE,
++
++    /**
++     * A gossip with a minor positive effect. This is only gained through curing a zombie
++     * villager.
++     */
++    MINOR_POSITIVE,
++
++    /**
++     * A gossip with a major positive effect. This is only gained through curing a zombie
++     * villager.
++     */
++    MAJOR_POSITIVE,
++
++    /**
++     * A gossip with a minor positive effect. This is only gained through trading with a villager.
++     */
++    TRADING,
++}
+diff --git a/src/main/java/org/bukkit/entity/Villager.java b/src/main/java/org/bukkit/entity/Villager.java
+index ef48ad9b..c5812e37 100644
+--- a/src/main/java/org/bukkit/entity/Villager.java
++++ b/src/main/java/org/bukkit/entity/Villager.java
+@@ -1,10 +1,13 @@
+ package org.bukkit.entity;
+ 
+ import java.util.Locale;
++import java.util.Map;
++import java.util.UUID;
+ import org.bukkit.Keyed;
+ import org.bukkit.Location;
+ import org.bukkit.NamespacedKey;
+ import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
```

Comment:
Shouldn't there be //Paper comments in here or something? I think there should be, but I didn't check.

Reply:
Should be fixed now.